cmd/abapEnvironmentCloneGitRepo.go

```diff
@@ -71,33 +67,39 @@ func runAbapEnvironmentCloneGitRepo(config *abapEnvironmentCloneGitRepoOptions,
 
 	log.Entry().Infof("Start cloning %v repositories", len(repositories))
 	for _, repo := range repositories {
+
+		commitString := ""
+		if repo.CommitID != "" {
+			commitString = ", commit '" + repo.CommitID + "'"
+		}
+
 		log.Entry().Info("-------------------------")
-		log.Entry().Info("Start cloning " + repo.Name + " with branch " + repo.Branch)
+		log.Entry().Info("Start cloning " + repo.Name + ", branch " + repo.Branch + commitString)
 		log.Entry().Info("-------------------------")
 
 		// Triggering the Clone of the repository into the ABAP Environment system
-		uriConnectionDetails, errorTriggerClone := triggerClone(repo.Name, repo.Branch, connectionDetails, client)
+		uriConnectionDetails, errorTriggerClone := triggerClone(repo, connectionDetails, client)
 		if errorTriggerClone != nil {
-			return errors.Wrapf(errorTriggerClone, "Clone of '%s' with branch '%s' failed on the ABAP System", repo.Name, repo.Branch)
+			return errors.Wrapf(errorTriggerClone, "Clone of '%s', branch '%s'%s failed on the ABAP System", repo.Name, repo.Branch, commitString)
 		}
 
 		// Polling the status of the repository import on the ABAP Environment system
 		status, errorPollEntity := abaputils.PollEntity(repo.Name, uriConnectionDetails, client, com.GetPollIntervall())
 		if errorPollEntity != nil {
-			return errors.Wrapf(errorPollEntity, "Clone of '%s' with branch '%s' failed on the ABAP System", repo.Name, repo.Branch)
+			return errors.Wrapf(errorPollEntity, "Clone of '%s', branch '%s'%s failed on the ABAP System", repo.Name, repo.Branch, commitString)
 		}
 		if status == "E" {
-			return errors.New("Clone of '" + repo.Name + "' with branch '" + repo.Branch + "' failed on the ABAP System")
+			return errors.New("Clone of '" + repo.Name + "', branch '" + repo.Branch + "'" + commitString + " failed on the ABAP System")
```

Comment:
Minor comment: IMO this further reduces the readability of the `runAbapEnvironmentCloneGitRepo` method. Looking at the other step implementations I would prefer to introduce a helper function iterating over the repositories and performing the Pull/Clone/Checkout operations rather than doing everything on top level here.

Furthermore I think we could refactor such a helper method performing a given operation over a list of repositories where we had a discussions previously. But this is definitely not necessary and doesn't belong to the scope of this PR.